app/web_conn.go

```diff
@@ -169,6 +171,11 @@ func (wc *WebConn) writePump() {
 		wc.WebSocket.Close()
 	}()
 
+	var buf bytes.Buffer
+	// 2k is seen to be a good heuristic under which 98.5% of message sizes remain.
+	buf.Grow(1024 * 2)
+	enc := json.NewEncoder(&buf)
```

Comment:
Sorry didn't get this, where is an array coming from?

Reply:
I was just wondering if using an array (e.g. `var data [2048]byte`) as the base of the buffer would be more efficient that allocating the data on the heap which is what `buf.Grow` does. But I don't know if that would escape to the heap anyway so that's why I was asking.

Reply:
Ahh gotcha, let me check that.

Reply:
Does not make much difference. The thing is that we have to change the array to a slice anyways when creating `NewBuffer`. So it escapes in both cases. On the contrary, `var buf bytes.Buffer` does not escape, whereas `bytes.NewBuffer` does.

```go
func main() {
	// var data [2048]byte
	// buf := bytes.NewBuffer(data[:])

	var buf bytes.Buffer
	buf.Grow(2048)

	enc := json.NewEncoder(&buf)
	err := enc.Encode(struct{ Name string }{Name: "hi"})
	if err != nil {
		fmt.Println(err)
	}
	fmt.Print(buf.String())
}
```

Tried 2 times. Once with the commented code, and later with the new code below.

First time:
```
proxy.go:10:6: moved to heap: data
proxy.go:11:24: &bytes.Buffer literal escapes to heap
proxy.go:13:24: &json.Encoder literal does not escape
```

Second time:
```
proxy.go:14:10: main ignoring self-assignment in bytes.b.buf = bytes.b.buf[:bytes.m·3]
proxy.go:13:6: moved to heap: buf
proxy.go:16:24: &json.Encoder literal does not escape
```